images/prow-tests/Dockerfile

```diff
@@ -12,42 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM gcr.io/k8s-testimages/kubekins-e2e:v20210113-cc576af-master AS stable-stuff
-
-ENV DEBIAN_FRONTEND noninteractive
+# Starting from the gcloud debian image https://github.com/GoogleCloudPlatform/cloud-sdk-docker/blob/master/debian_slim/Dockerfile
 ARG GCLOUD_VERSION=368.0.0
-ARG KUBECTL_VERSION=v1.21.4
 
-# If we don't run update, the apt-get install below don't work
-RUN apt-get update
+FROM google/cloud-sdk:${GCLOUD_VERSION}-slim AS stable-stuff
 
-# gcloud
-RUN gcloud components update --version=${GCLOUD_VERSION}
+ENV DEBIAN_FRONTEND noninteractive
+
+ARG KUBECTL_VERSION=v1.21.4
+ARG BAZELISK_VERSION=v1.11.0
+ARG DOCKER_VERSION=5:20.10.9~3-0~debian-buster
 
 # kubectl
-RUN rm -f "$(which kubectl)" && \
-    wget "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" -O /usr/local/bin/kubectl && \
+RUN curl -fsSL "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" -o /usr/local/bin/kubectl && \
     chmod +x /usr/local/bin/kubectl
 
-# Extra tools through apt-get
-RUN apt-get install -y uuid-runtime   # for uuidgen
-RUN apt-get install -y shellcheck     # for shellcheck presubmit
+# bazel
+RUN curl -fsSL "https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-linux-amd64" -o /usr/local/bin/bazel && \
+    chmod +x /usr/local/bin/bazel
+
+# Docker including docker-ce, docker-ce-cli, and containerd.io
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - && \
+    echo "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list && \
+    apt-get update && \
+    apt-get -y install --no-install-recommends docker-ce="${DOCKER_VERSION}" docker-ce-cli="${DOCKER_VERSION}"
 
 # Setup adoptopenjdk
 ARG JAVA_VERSION=16
-RUN apt-get install -y wget apt-transport-https gnupg
-RUN wget -qO - https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public | apt-key add -
+RUN curl -sf https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public | apt-key add -
```

Comment:
What's the reason for changing `wget` to `curl`?

Reply:
So I don't have to install wget. curl is already bundled

Reply:
I did some search and it seems at least serving is using `wget`. So let's keep it?

Reply:
i'll add it back in